### ISSUE TITLE
Key Repeat:::: Duplicate keys  with  10..15.0

### ISSUE THREAD

**Issue by steakknife** [tekezo/Karabiner#539]
10.15.0  is broken,,,,   system unusable.

Delay     100 ms

Repat      10 ms

Woooorkked fineee   with   10.1444....0

OS X::  10.11.2 (15C50)

**Comment by tekezo** [comment]
The safety limit of key repeat rate has been removed at v10.15.0.
(Delay is limited to 200 ms at v10.14.0 even if you specified smaller value.)

Please increase Delay to 200 ms in v10.15.0.

**Comment by trevordmiller** [comment]
I was having this same problem of repeating characters (- became -----, " " became "      " etc.). Increasing delay to 200ms as @tekezo suggested fixed this for me. Thank you!

**Comment by steakknife** [comment]
This is an extremely disappointing change. The usual UX is to replace the
field value with a reasonably safe minimum, not throw out usability
altogether and create a new landmine.

**Comment by tekezo** [comment]
Sorry for inconvenience.

I assumed very few users are using the exterme fast key repeat values,
and I decided to remove the limit because we are enable to adjust the values via GUI with mouse.

I'll be more careful when I introduce an incompatible change next time.

**Comment by ethus3h** [comment]
Maybe could detect similar issue and just show a message box "Would you like to set value to 200 since that's what it's effectively been so far? Or would you like to use value you specified?"...

**Comment by tekezo** [comment]
I wrote similar message in the update message but the most people did not read...
I'm afraid that the dialog is also ignored.

**Comment by ethus3h** [comment]
Oh well. :(

**Comment by ethus3h** [comment]
I would think this issue can be closed now, then, as it seems to be a case of everything that could be done has been done?

**Comment by rjhilgefort** [comment]
:+1: I experienced this issue as well- quite frustrating. Happy to find this issue though. Maybe call out these sort of changes more obviously in the update notes.

**Comment by steakknife** [comment]
# TL;DR

It goes to this [SO article](http://sqa.stackexchange.com/questions/2856/release-notes-best-practices) on *release notes* (short) vs. *release contents* (full).

The Sparkle release notes seems like a good place for news... *concise* highlights list of what's different.

If it were me, I would consider tightening up copy and moving full details to a changelog page.

![img3](http://content.screencast.com/users/xzxzxzxx/folders/Jing/media/1f447cc6-6854-4398-915b-29c97387a1eb/00000033.png)

---
## Mock up

![example](http://content.screencast.com/users/xzxzxzxx/folders/Jing/media/8deb4729-9a34-44bc-9a4e-ce17e877447d/00000038.png)

[Source here](http://codepen.io/anon/pen/gPmqJG) of a mocked up, feel free to rip-off / get inspired / etc.

--- 
## Other examples

### TotalFinder

![img2](http://content.screencast.com/users/xzxzxzxx/folders/Jing/media/668c1e64-08f3-4361-ab87-9f23a2a37ba9/00000036.png)

### ExpanDrive

![img](http://content.screencast.com/users/xzxzxzxx/folders/Jing/media/4eec0d00-d655-4297-92b3-ddd151d72cd5/00000034.png)

[Here's ExpanDrive's update dialog contents](http://updates.expandrive.com/apps/expandrive/v/5-1-6/release_notes)

^--- note: redundant verbs

The server-side [appcast.xml](https://pqrs.org/osx/karabiner/files/appcast.xml) contains the html of what's rendered in the update dialog,

**Comment by tekezo** [comment]
OK.
I updated message style.
![160107-0002](https://cloud.githubusercontent.com/assets/659178/12168768/c688ca9a-b577-11e5-86e1-d495209a17e7.png)

**Comment by steakknife** [comment]
¡Perfectamente!
